src/main/java/com/uber/kafkaSpraynozzle/KafkaPoster.java

```diff
@@ -17,30 +20,49 @@
 import org.apache.http.util.EntityUtils;
 
 public class KafkaPoster implements Runnable {
-    private static int currentUrl = 0;
-
     private MetricRegistry metricRegistry;
-    private ConcurrentLinkedQueue<ByteArrayEntity> queue;
-    private PoolingHttpClientConnectionManager cm;
-    private List<String> urls;
-    private KafkaFilter messageFilter;
+    ConcurrentLinkedQueue<ByteArrayEntity> queue;
+    PoolingHttpClientConnectionManager cm;
+    List<String> urls;
+    int currentUrl = 0;
+    static final int responseDecayingHalflife = 5;
+    KafkaFilter messageFilter;
     private RequestConfig requestConfig;
-
-    private Timer postTime;
-    private Counter postCount;
-    private Counter postSuccess;
-    private Counter postFailure;
-    private Counter filteredCount;
+    private boolean roundRobin = false;
+    private int batchSize = 20;
+    private int batchMaxBytes = 128*1024;
+    private boolean debuggingOutput = false;
+    private Timer postTime = new Timer();
+    private Counter postCount = new Counter();
```

Comment:
Yes that would be less ugly. I have updated the diff to avoid passing in nulls; but somehow I think here keeping the code like this is also not bad.